src/Access/SuperUserAccessCheck.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Check whether a user has the uid 1.
+ */
+class SuperUserAccessCheck implements AccessInterface {
+
+  /**
+   * Checks access.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The logged in user key.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(AccountInterface $account): AccessResultInterface {
+    $uid = $account->id();
```

Comment:
`$uid` is only used only once, therefore I would not create a variable for it and refactor the code a bit.

```php
    if (1 === $account->id()) {
```

Regarding the missing empty line before the return, as there is no rule in PHPCS, I wouldn't flag this for now but highlight this during the next retro.

As this is not a blocker nor defined in any PHPCS, it's up to the user to decide if he wants to do it or not, it's not mandatory.